mergify_engine/utils.py

```diff
@@ -108,54 +110,66 @@ def compute_hmac(data):
     return str(mac.hexdigest())
 
 
+@dataclasses.dataclass
 class Gitter(object):
-    def __init__(self, logger):
-        self.tmp = tempfile.mkdtemp(prefix="mergify-gitter")
-        self.logger = logger
+    logger: logging.LoggerAdapter
+    tmp: str = dataclasses.field(init=False)
+
+    async def init(self):
+        self.tmp = await asyncio.to_thread(tempfile.mkdtemp, prefix="mergify-gitter")
         self.logger.info("working in: %s", self.tmp)
+        await self("init")
 
-    def __call__(self, *args, **kwargs):  # pragma: no cover
+    async def __call__(self, *args, **kwargs):  # pragma: no cover
         self.logger.info("calling: %s", " ".join(args))
         kwargs["cwd"] = self.tmp
-        # Worker timeout at 5 minutes, so ensure subprocess return before
-        kwargs["timeout"] = 4 * 60 + 30
         kwargs["encoding"] = "utf-8"
-        kwargs.setdefault("check", True)
-        kwargs.setdefault("stdout", subprocess.PIPE)
-        kwargs.setdefault("stderr", subprocess.STDOUT)
+        kwargs.setdefault("stdout", asyncio.subprocess.PIPE)
+        kwargs.setdefault("stderr", asyncio.subprocess.STDOUT)
         try:
-            return subprocess.run(["git"] + list(args), **kwargs)
+            p = await asyncio.create_subprocess_exec(
+                "git", ["git"] + list(args), **kwargs
+            )
+            # Worker timeout at 5 minutes, so ensure subprocess return before
+            await asyncio.wait_for(p.wait(), 4 * 60 + 30)
+            if p.returncode:
+                raise subprocess.CalledProcessError(
```

Comment:
should we stop using `subprocess` entirely here?

Reply:
I reviewed the API of gitter (exception and returned value), move it into a separate file, added tests.